### ISSUE TITLE
[BUG] Wear system

### ISSUE THREAD

**Issue by crashzk** [Nereziel/cs2-WeaponPaints#125]
Some skins, even though they are configured as new from the factory, still appear a little worn out.

![20240128133355_1](https://github.com/Nereziel/cs2-WeaponPaints/assets/32937653/26599982-aa27-48cb-83b9-dee7e7d5a84a)
![20240128133356_1](https://github.com/Nereziel/cs2-WeaponPaints/assets/32937653/0417ffc4-ba64-4214-9fd9-e2494cb7133a)
![20240128133413_1](https://github.com/Nereziel/cs2-WeaponPaints/assets/32937653/1ab7e239-8aff-4787-b39c-28398c34dfcf)

On the website I configured the **Desert Eagle | Blaze** like new from the factory.

![image](https://github.com/Nereziel/cs2-WeaponPaints/assets/32937653/dc2f34ea-e23f-4a09-baec-99c7480ed95f)

**Comment by Nereziel** [comment]
![obrazek](https://github.com/Nereziel/cs2-WeaponPaints/assets/8618825/77269f6e-1c30-47d3-91c3-2c556c0d4583)

**Comment by crashzk** [comment]
> [Build 168](https://github.com/Nereziel/cs2-WeaponPaints/releases/tag/build-168) + bklols ext

@Nereziel strange, I'm using version [Build 170](https://github.com/Nereziel/cs2-WeaponPaints/releases/tag/build-170)

MetaMod: 2.0.0-dev+1280
CSSharp: Build 159

I just didn't delete the database, would that be necessary?

**Comment by Nereziel** [comment]
try change wear in database to 0.0

**Comment by crashzk** [comment]
> try change wear in database to 0.0

Currently it looks like this:
![image](https://github.com/Nereziel/cs2-WeaponPaints/assets/32937653/f7a9dad0-44b1-4641-8d75-5a9eb366bdb8)

I will be changing it to `0.0` and testing, I will return here with results.

**Comment by crashzk** [comment]
More complicated to change than I thought... Due to the formatting that saves the data in the database by phpByAdmin, we cannot edit it "directly", so to speak. Just using SQL command to change the value and such.

![image](https://github.com/Nereziel/cs2-WeaponPaints/assets/32937653/a56623ff-cafb-4de5-a761-e63e5452b7d9)

I racked my brains here, but to no avail. Having said that, I believe we can make a suggestion for improvement.

Rewrite this database, better formatting, how it saves information and such.

In my case, it saves my steamid several times for each skin and weapon configuration. I believe that we can think of at least one option to edit without using the SQL command, it is easier for those who would not have so much knowledge in programming.

**Comment by Nereziel** [comment]
sorry my bad, that wouldnt help, zero is still zero :D

**Comment by crashzk** [comment]
> sorry my bad, that wouldnt help, zero is still zero :D

So, I chose to completely delete the data, reset the database... Plugin automatically set the skins wear to `0`, even trying to change it to `0.0` on the website doesn't work, it still saves as `0`.

![image](https://github.com/Nereziel/cs2-WeaponPaints/assets/32937653/21536ef3-4113-49e0-a990-f3a121554d4c)

The only way it saved was `0.01`, I believe the plugin could be set to `0.00` by default.

![image](https://github.com/Nereziel/cs2-WeaponPaints/assets/32937653/7455cc87-f3bb-4083-add3-4b61e6a2b356)

![20240130212128_1](https://github.com/Nereziel/cs2-WeaponPaints/assets/32937653/48b8ccda-4079-4ffa-a696-d12a6be9e8fa)

Anyway, can we expect an update, fix to automatically apply wear `0.01` or even `0.00` to all skins?

**Comment by Nereziel** [comment]
Its already existing, default float is .00001
https://github.com/Nereziel/cs2-WeaponPaints/blob/43e7a3183e435d1b91fcb2424c4d5c9491c1f399/Utility.cs#L42

**Comment by crashzk** [comment]
> Its already existing, default float is .00001
> 
> https://github.com/Nereziel/cs2-WeaponPaints/blob/43e7a3183e435d1b91fcb2424c4d5c9491c1f399/Utility.cs#L42

So we have a problem, bug. I created a new database, configured everything correctly, but whenever I choose **Desert Eagle | Blaze** without changing wear or seed still appears worn out, the wear marked as `0` and not `0.0001`

**Comment by crashzk** [comment]
I also noticed that if we try to set the wear to `0.000001` it shows several characters instead of numbers.

![image](https://github.com/Nereziel/cs2-WeaponPaints/assets/32937653/4f71ff5d-1bc6-4880-bc0a-1424d96620ca)

Would it be correct? To show correct wear, only three zeros after the dot are allowed `0.0001`.

![image](https://github.com/Nereziel/cs2-WeaponPaints/assets/32937653/77a5bd82-827b-479c-9f06-8426355afc44)

Above that it breaks and does not show as a number as you can see in the printout.

**Comment by crashzk** [comment]
> try with reworked db: https://github.com/Nereziel/cs2-WeaponPaints/releases/tag/dbtest-1

@Nereziel done, clean installation of the website and database, everything reset, everything in a new database. BUG still occurs, wear defaults to `0` instead of `0.000001` and the character BUG in wear continues as well.

![image](https://github.com/Nereziel/cs2-WeaponPaints/assets/32937653/cff9602f-6bf9-431e-97e0-cef3b198588f)

**Comment by crashzk** [comment]
> i have the same problem, but i think its because i have my own skin on that gun and the wear of that skin ports on the weaponpaints skin. When i unequip my skin in loadout the wear is the way i set it to be.

This happens even with the default CS2 skin.

**Comment by crashzk** [comment]
> try with reworked db: https://github.com/Nereziel/cs2-WeaponPaints/releases/tag/dbtest-1

And in this version, even leaving `GiveKnifeAfterRemove` to **true** the knives are still not appearing, they are removed anyway. At least with me.

**Comment by Nereziel** [comment]
Doe's newer versions fixed this?

**Comment by crashzk** [comment]
> Doe's newer versions fixed this?

No, I continue to have this bug with the most recent versions, even though I set the skin to **Factory New**, it still shows wear and tear.

And in the database the skins are saved as **wear 0** instead of `0.000001`

![image](https://github.com/Nereziel/cs2-WeaponPaints/assets/32937653/0fa3507f-3a43-4bd2-9fac-0237b9e60760)

**Comment by crashzk** [comment]
Just for the record, newer version the bug still occurs:
```
  [#9:LOADED]: "WeaponPaints" (2.2b) by Nereziel & daffyy
  Skin, gloves, agents and knife selector, standalone and web-based
```

![20240306154210_1](https://github.com/Nereziel/cs2-WeaponPaints/assets/32937653/3b99dd7f-f33a-4409-8693-33b1a850939c)

![20240306154219_1](https://github.com/Nereziel/cs2-WeaponPaints/assets/32937653/c089ba87-b944-4d36-9a11-9a5ab003cb3f)

![20240306154223_1](https://github.com/Nereziel/cs2-WeaponPaints/assets/32937653/18329fe9-6a13-43b7-bc94-3737209777f5)

**Comment by daffyyyy** [comment]
Still exist?
